crypto/src/key/rschnorr/mod.rs

```diff
@@ -155,6 +162,105 @@ impl Derivable for MLRistrettoPrivateKey {
     }
 }
 
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub struct MLPrivateKey {
+    key_data: secp256k1::SecretKey,
+}
+
+impl zeroize::Zeroize for MLPrivateKey {
+    fn zeroize(&mut self) {
+        // TODO find a safe way to retrieve a mutable slice of the underlying SecretKey data array
+        let key_data = unsafe {
+            std::slice::from_raw_parts_mut(
+                self.key_data.as_mut_ptr(),
+                secp256k1::constants::SECRET_KEY_SIZE,
+            )
+        };
+        key_data.zeroize();
+    }
+}
+
+impl Drop for MLPrivateKey {
```

Comment:
I think you don't need to implement this... just inherit Zeroize and that'll do it.